portal/models/user.py

```diff
@@ -1605,6 +1605,40 @@ def delete_user(self, acting_user):
         Token.query.filter_by(user_id=self.id).delete()
         db.session.commit()
 
+    def reactivate_user(self, acting_user):
+        """Reactivate a previously deleted user
+
+        This method clears the deleted status - by removing the link from
+        the user to the audit recording the delete.  Audit itself is retained
+        for tracking purposes, and a new one will be created for posterity
+
+        :param self: user to reactivate
+        :param acting_user: individual executing the command, for audit trail
+
+        """
+        if not self.deleted:
+            raise ValueError("can't reactivate active user")
+        if self == acting_user:
+            raise ValueError("can't delete self")
+        if acting_user is None:
+            raise ValueError("reactivate requires well defined acting_user")
+
+        # The email was masked during delete.  Need to confirm a user didn't
+        # sneak in with the same address while deleted.  The accessor returns
+        # the unmasked value.
+        unmasked = self.email
+        if User.query.filter(User.email == unmasked).count() > 0:
```

Comment:
Do we need to consider (lower)casing here?

Reply:
aye!